### PR TITLE
JARM (JWT Secured Authorization Response Mode) for OAuth 2.0 implementation

### PR DESCRIPTION
**Pull request by anjuchamantha**
# JARM Implementation

Fixes the issue: https://github.com/wso2/product-is/issues/15527

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java:4]:
```suggestion
 * WSO2 LLC. licenses this file to you under the Apache License,
```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java:1]: Check the license header in all newly added files and update as suggested below.

Comment (anjuchamantha) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/responsemode/provider/AbstractResponseModeProvider.java:1]: Updated

**Review by chamathns — APPROVED** [review]

**Pull request merged**